l10n_ch_invoice_reports/models/account_invoice.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+
+from odoo import api, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    # override to add context key dependency and handle it properly inside method
+    @api.depends(
+        "type",
+        "name",
+        "currency_id.name",
+        "invoice_partner_bank_id.l10n_ch_isr_subscription_eur",
+        "invoice_partner_bank_id.l10n_ch_isr_subscription_chf",
+    )
+    @api.depends_context("_mail_template_no_attachments")
+    def _compute_l10n_ch_isr_valid(self):
+        """Returns True if all the data required to generate the ISR are present"""
+        for record in self:
+            if self.env.context.get("_mail_template_no_attachments", False):
+                record.l10n_ch_isr_valid = False
+                return False
```

Comment:
This `return` statement should not be there. Because all records will not be computed.